widget/edit.controller.js

```diff
@@ -30,15 +32,26 @@
         entity.loadFields().then(function () {
           $scope.fieldsArray = entity.getFormFieldsArray();
           $scope.fieldsArray = _.filter($scope.fieldsArray, function(item){ return item.model === 'picklists'; });
-          if(config.picklistItem){
+          if($scope.config.picklistItem){
             var selectedPicklistItem = _.find($scope.fieldsArray, function(item){
-            return item.name === config.picklistItem;
+            return item.name === $scope.config.picklistItem;
             });
             $scope.loadlistitem(selectedPicklistItem);
           }
-        }
-                                );
+        });
       }
+
+      $scope.loadListItemValuesToUpdate = function (picklistItem) {
```

Comment:
The same code block is used from lines 65 to 72. That can be replaced with this function call.